File: backend/SRMS.API/Models/CctvNode.cs
namespace SRMS.API.Models;

public class CctvNode
{
    public int Id { get; set; }
    public int NodeId { get; set; }
    public string CameraName { get; set; } = string.Empty;
    public string Location { get; set; } = string.Empty;
    public string CameraType { get; set; } = string.Empty;      // LAPTOP_WEBCAM | MOBILE_CAMERA | NETWORK_STREAM
    public string Status { get; set; } = "OFFLINE";             // OFFLINE | CONNECTING | ONLINE | ERROR | ANALYZING
    public string StreamSource { get; set; } = string.Empty;    // LOCAL_WEBCAM | NETWORK_STREAM | PHONE_CAMERA
    public string StreamUrl { get; set; } = string.Empty;       // configurable network stream URL (no secrets)
    public string CreatedAt { get; set; } = string.Empty;
    public string LastSeenAt { get; set; } = string.Empty;

    // Backend-provided CCTV location (never the phone GPS).
    public double? Lat { get; set; }
    public double? Lng { get; set; }
}

File: web/src/modules/crime-vehicle/models/cctvNode.js
export const CAMERA_TYPES = ['LAPTOP_WEBCAM', 'MOBILE_CAMERA', 'NETWORK_STREAM'];
export const STREAM_SOURCES = ['LOCAL_WEBCAM', 'PHONE_CAMERA', 'NETWORK_STREAM'];
export const NODE_STATUSES = ['OFFLINE', 'CONNECTING', 'ONLINE', 'ERROR', 'ANALYZING'];

export const normalizeNode = (raw) => ({
  id: raw?.id ?? raw?.nodeId ?? 0,
  nodeId: raw?.nodeId ?? raw?.id ?? 0,
  cameraName: raw?.cameraName || raw?.name || `CCTV Node ${raw?.nodeId ?? ''}`,
  location: raw?.location || raw?.zone || 'Unknown location',
  cameraType: raw?.cameraType || 'NETWORK_STREAM',
  status: NODE_STATUSES.includes(raw?.status) ? raw.status : 'OFFLINE',
  streamSource: raw?.streamSource || 'NETWORK_STREAM',
  streamUrl: raw?.streamUrl || '',
  createdAt: raw?.createdAt || '',
  lastSeenAt: raw?.lastSeenAt || 'Never'
});

export const phoneStreamUrl = (node) => {
  if (node?.streamUrl) return node.streamUrl;
  return import.meta.env.VITE_PHONE_CAMERA_URL || '';
};

export const isPhoneNode = (node) =>
  node?.streamSource === 'PHONE_CAMERA' || node?.cameraType === 'MOBILE_CAMERA';

export const isWebcamNode = (node) =>
  node?.streamSource === 'LOCAL_WEBCAM' || node?.cameraType === 'LAPTOP_WEBCAM';
